src/utilities/include/structs.h

```diff
@@ -222,8 +222,13 @@ struct Curvature {
  * Mesh.
  */
 struct BaryRef {
-  /// The original mesh triangle index.
+  /// The unique ID of the mesh instance of this triangle. If .meshID and .tri
+  /// match for two triangles, then they are coplanar and came from the same
+  /// face.
   int meshID;
+  /// The OriginalID of the mesh this triangle came from. This ID is ideal for
+  /// reapplying properties like UV coordinates to the output mesh.
+  int originalID;
```

Comment:
Here's the other API change: now we publicly store both the originalID (what meshID was before) and the meshID, which is now a mesh instance ID. The nice thing about exposing this, is that you can tell if triangles can be merged into the same polygon if their `meshID` and `tri` match, which is the same system used by our decimator.